core.cc

```diff
@@ -90,10 +93,12 @@ NAN_METHOD(SysLog) {
 
     if (message) {
         Nan::AsyncQueueWorker(new Worker(callback, priority, message));
-    } else if(!callback->IsEmpty()) {
+        callback = NULL; // Memory owned by Worker
+    } else if(info[2]->IsFunction()) {
         callback->Call(0, NULL);
     }
 
+    delete callback;
```

Comment:
I NULL callback in the if block, so it can be deleted no matter whether it goes thru if or else, but I can rearrange.